src/Runner.Listener/JobDispatcher.cs

```diff
@@ -531,11 +544,8 @@ private async Task RunAsync(Pipelines.AgentJobRequestMessage message, string orc
                                     detailInfo = string.Join(Environment.NewLine, workerOutput);
                                     Trace.Info($"Return code {returnCode} indicate worker encounter an unhandled exception or app crash, attach worker stdout/stderr to JobRequest result.");
 
-                                    var jobServer = HostContext.GetService<IJobServer>();
-                                    VssCredentials jobServerCredential = VssUtil.GetVssCredential(systemConnection);
-                                    VssConnection jobConnection = VssUtil.CreateConnection(systemConnection.Url, jobServerCredential);
-                                    await jobServer.ConnectAsync(jobConnection);
 
+                                    var jobServer = await InitializeJobServerAsync(systemConnection);
```

Comment:
This will give back a run server or job server (like today) based on the message